pulsar-client-cpp/lib/PartitionedProducerImpl.cc

```diff
@@ -37,18 +39,28 @@ namespace pulsar {
                                                      const ProducerConfiguration& config):client_(client),
                                                                                           destinationName_(destinationName),
                                                                                           topic_(destinationName_->toString()),
-                                                                                          numPartitions_(numPartitions),
                                                                                           conf_(config),
                                                                                           state_(Pending)
     {
         numProducersCreated_ = 0;
+
+        topicMetadata_ = std::unique_ptr<TopicMetadata>(new TopicMetadataImpl(numPartitions));
```

Comment:
In general we cannot yet use all the c++11 feature in general lib. We could use a `boost::scoped_ptr` here.